### PR TITLE
Added loop around directory listing call to retry the call, to work a…

### PR DESCRIPTION
**Pull request by kevinkrieger**
…round the timeouts issue.

### CODE REVIEW

File: sync_radar_data_globus.py [hunk 1]
```diff
@@ -121,7 +122,7 @@ def __init__(self, client_id, client_secret=None, transfer_rt=None):
         self.sync_pattern = args.sync_pattern
         self.data_type = args.data_type
         self.sync_local_dir = args.sync_local_dir
-        self.mirror_root_dir = "~/chroot/sddata"
+        self.mirror_root_dir = "/chroot/sddata"
```
Comment (mts299) [inline sync_radar_data_globus.py:134]: Sorry, can you please explain why you took out `~` (in this comment, I am more just curious if something changed with cedar vs. globus)?

File: sync_radar_data_globus.py [hunk 2]
```diff
@@ -171,8 +172,37 @@ def synchronize(self):
                 pass
             else:
                 pass
-            listing = self.transfer_client.operation_ls(self.mirror_uuid, path=listing_path,
-                                                        filter=listing_pattern)
+            print("Listing path: {path} on endpoint: {endpoint} with pattern: {pattern}".format(
+                path=listing_path, endpoint=self.mirror_uuid, pattern=listing_pattern))
+            listing_succeeded = False
+            operation_ls_retries = 10
```
Comment (mts299) [inline sync_radar_data_globus.py]: Any reason for 10?

File: sync_radar_data_globus.py [hunk 3]
```diff
@@ -171,8 +172,37 @@ def synchronize(self):
...
+            while attempts < operation_ls_retries:
+                try:
+                    listing_times.append(time.time())
+                    print("Attempt: {}".format(attempts)),
+                    listing = self.transfer_client.operation_ls(self.mirror_uuid, path=listing_path,
+                                                                filter=listing_pattern)
+                    listing_succeeded = True
+                    break
+                except globus_sdk.GlobusAPIError:
+                    print("... failed")
```
Comment (mts299) [inline sync_radar_data_globus.py]: Would it be helpful to `print('...successful')` if it succeeds as well?

File: sync_radar_data_globus.py [hunk 4]
```diff
@@ -171,8 +172,37 @@ def synchronize(self):
...
+                    listing_succeeded = False
+                    attempts += 1
+
+            for index, listing_time in enumerate(listing_times):
```
Comment (mts299) [inline sync_radar_data_globus.py]: Can you add a comment on what this looping is doing? 
If we are successful, then is this information still useful?

File: sync_radar_data_globus.py [hunk 5]
```diff
@@ -171,8 +172,37 @@ def synchronize(self):
...
+                if index == 0:
+                    print(listing_time)
```
Comment (mts299) [inline sync_radar_data_globus.py]: A more informative print statement might be more clear to the user as well. 
"Attempt 1 took this many minutes {time}"

File: sync_radar_data_globus.py [hunk 6]
```diff
@@ -171,8 +172,37 @@ def synchronize(self):
...
+                    continue
+                else:
+                    print(listing_time-listing_times[index-1])
```
Comment (mts299) [inline sync_radar_data_globus.py]: Again what is this value going to mean to the user when it prints out?

File: sync_radar_data_globus.py (hunk 1 above)
Comment (kevinkrieger) [reply sync_radar_data_globus.py:134]: It changed when it went from graham to cedar, not sure why...

File: sync_radar_data_globus.py (hunk 2 above)
Comment (kevinkrieger) [reply sync_radar_data_globus.py]: it was the number suggested by Sergiy, but I've found that 15 is more reasonable, so I've changed it to 15. I'll add comment

File: sync_radar_data_globus.py (hunk 3 above)
Comment (kevinkrieger) [reply sync_radar_data_globus.py]: I think I'll take out the ...failed statement and only print a dot every time it retries, then print 'failed' if it runs out of retries,

File: sync_radar_data_globus.py (hunk 4 above)
Comment (kevinkrieger) [reply sync_radar_data_globus.py]: Decided to get rid of this, it was really only for testing.

File: sync_radar_data_globus.py (hunk 5 above)
Comment (kevinkrieger) [reply sync_radar_data_globus.py]: decided to get rid of this loop

File: sync_radar_data_globus.py (hunk 6 above)
Comment (kevinkrieger) [reply sync_radar_data_globus.py]: decided to get rid of this loop

File: sync_radar_data_globus.py (hunk 3 above)
Comment (mts299) [reply sync_radar_data_globus.py]: Sounds good to me, what about success?

File: sync_radar_data_globus.py [hunk 7]
```diff
@@ -174,32 +183,37 @@ def synchronize(self):
...
+            operation_ls_retries = 15
             attempts = 0
             listing = []
             listing_times = []
 
             while attempts < operation_ls_retries:
                 try:
                     listing_times.append(time.time())
-                    print("Attempt: {}".format(attempts)),
+                    if PYTHON3:
```
Comment (mts299) [inline sync_radar_data_globus.py:199]: Any reason to change the printing format method? I thought `.format` works in 2.7?

File: sync_radar_data_globus.py [hunk 8]
```diff
@@ -38,13 +38,20 @@
...
 import inspect
 from datetime import datetime
 from os.path import expanduser, isfile
 import sys
 import argparse
 import globus_sdk
 import time
+import sys
+
+if sys.version_info > (3, 0):
```
Comment (mts299) [inline sync_radar_data_globus.py]: Do you want `>= (3, 0)`?

File: sync_radar_data_globus.py (hunk 3 above)
Comment (kevinkrieger) [reply sync_radar_data_globus.py]: I think that success is implied if it moves on, so don't think it's necessary

File: sync_radar_data_globus.py (hunk 8 above)
Comment (kevinkrieger) [reply sync_radar_data_globus.py]: good point, I'll change it

File: sync_radar_data_globus.py (hunk 7 above)
Comment (kevinkrieger) [reply sync_radar_data_globus.py:199]: Didn't change, I just decided to remove the print statement with number of attempts.

**Comment by mts299** [comment]
@kevinkrieger there is conflicts with this branch, can you resolve them so I can merge it?

**Pull request merged**